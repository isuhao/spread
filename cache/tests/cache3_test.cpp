#include <iostream>
#include <boost/filesystem.hpp>

#include "index.hpp"

using namespace Spread;
using namespace std;
using namespace boost::filesystem;

Cache::CacheIndex cache;

Hash hello("hello",5);

int main()
{
  // Regression test

  copy_file("hello.dat", "_t3_out.dat");

  cache.addFile("_t3_out.dat");
  cache.addFile("hello.dat");

  cout << cache.findHash(hello) << endl;

  remove("_t3_out.dat");

  // FIXED segfault here
  cout << cache.findHash(hello) << endl;

  return 0;
}
