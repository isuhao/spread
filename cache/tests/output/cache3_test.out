/home/mortennk/koding/spread/cache/tests/_t3_out.dat
/home/mortennk/koding/spread/cache/tests/hello.dat
